gallery/gallery/lib/demos/material/picker_demo.dart

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+import 'package:intl/intl.dart';
+
+// BEGIN pickerDemo
+
+enum PickerDemoType {
+  date,
+  time,
+}
+
+class PickerDemo extends StatefulWidget {
+  const PickerDemo({Key key, this.type}) : super(key: key);
+
+  final PickerDemoType type;
+
+  @override
+  _PickerDemoState createState() => _PickerDemoState();
+}
+
+class _PickerDemoState extends State<PickerDemo> {
+  DateTime _fromDate = DateTime.now();
+  TimeOfDay _fromTime = TimeOfDay.fromDateTime(DateTime.now());
+
+  String get _title {
+    switch (widget.type) {
+      case PickerDemoType.date:
+        return GalleryLocalizations.of(context).demoDatePickerTitle;
+      case PickerDemoType.time:
+        return GalleryLocalizations.of(context).demoTimePickerTitle;
+    }
+    return '';
+  }
+
+  String get _labelText {
+    switch (widget.type) {
+      case PickerDemoType.date:
+        return DateFormat.yMMMd().format(_fromDate);
```

Comment:
does this require localization? Is this even possible to localize?

Reply:
This should work out of the box, the `DateFormat` class comes from the intl library, it's currently not working and I think https://github.com/flutter/flutter/pull/49357 will fix it.